lua/luasnip/loaders/from_snipmate.lua

```diff
@@ -0,0 +1,201 @@
+
+		body = table.concat(body, "\n")
+		local snip = ls.parser.parse_snippet({
+			trig = prefix,
+			dscr = description,
+			wordTrig = true,
+		}, body)
+		table.insert(snippet, snip)
+	end
+
+	while i <= #lines do
+		local line = lines[i]
+		if vim.startswith(line, "snippet") then
+			_parse()
+		elseif vim.startswith(line, "extends") then
+			local scopes = vim.split(vim.trim(line:sub(8)), "[,%s]+")
+			vim.list_extend(extends, scopes)
+			i = i + 1
+		elseif vim.startswith(line, "#") or line:find("^%s*$") then
+			-- comment and blank line
+			i = i + 1
+		else
+			error(("invalid line in %s: %s"):format(filename, i))
+		end
+	end
+
+	return snippet, extends
+end
+
+local function load_snippet_file(lang, path)
+	if not Path.exists(path) then
+		return
+	end
+
+	Path.async_read_file(
+		path,
+		vim.schedule_wrap(function(buffer)
+			local snippet, extends = parse_snipmate(buffer, path)
+			if not snippet then
+				return
+			end
+			table.insert(extends, lang)
+			for _, ft in ipairs(extends) do
+				local lang_snips = ls.snippets[ft] or {}
+				ls.snippets[ft] = vim.list_extend(lang_snips, snippet)
+				session.latest_load_ft = ft
+				vim.cmd("do User LuasnipSnippetsAdded")
+			end
+		end)
+	)
+end
+
+local function filter(exclude, include)
+	vim.validate({
+		exclude = { exclude, "table" },
+		include = { include, "table", true },
+	})
+
+	exclude = loader_util.filetypelist_to_set(exclude)
+	include = loader_util.filetypelist_to_set(include)
+
+	return function(lang)
+		if exclude[lang] then
+			return false
+		end
+		if include == nil or include[lang] then
+			return true
+		end
+	end
+end
+
+local function get_paths(root)
+	local ret = {}
+	local files = Path.scandir(root, "file")
+	for _, file in ipairs(files) do
+		local name, ext = Path.basename(file, true)
+		if ext == "snippets" then
+			table.insert(ret, { path = Path.join(root, file), ft = name })
+		end
+	end
+	local dirs = Path.scandir(root, "directory")
+	for _, dir in ipairs(dirs) do
+		local name = dir
+		files = Path.scandir(Path.join(root, dir), "file")
+		for _, file in ipairs(files) do
+			if vim.endswith(file, ".snippets") then
+				table.insert(
+					ret,
+					{ path = Path.join(root, dir, file), ft = name }
+				)
+			end
+		end
+	end
+	return ret
+end
+
+local function load_snippet_folder(root, opts)
+	local paths = get_paths(root)
+	local ft_filter = filter(opts.exclude, opts.include)
+
+	for _, v in ipairs(paths) do
+		local path, ft = v.path, v.ft
+		if ft_filter(ft) then
+			load_snippet_file(ft, path)
+		end
+	end
+end
+
+local M = {}
+
+function M.load(opts)
+	opts = opts or {}
+	opts.exclude = opts.exclude or {}
+
+	if not opts.paths then
+		opts.paths = vim.api.nvim_get_runtime_file("snippets", true)
+	elseif type(opts.paths) == "string" then
+		opts.paths = vim.split(opts.paths, ",")
+	end
+
+	opts.paths = vim.tbl_map(Path.expand, opts.paths)
+	opts.paths = util.deduplicate(opts.paths)
+
+	ls.filetype_extend("all", vim.g.luasnip_snipmate_global or { "_" })
```

Comment:
TBH I would put a notice into docs/examples saying that the `all`-filetype has to be manually extended instead of doing it here. That way the users are familiar with that mechanism and can use it on other sources.
Also, calling it multiple times will make the `ft_redirect`-table contain `_` multiple times, leading to slightly slower expand if no snippet matches (the table for `_` will be searched multiple times).

Reply:
Oh, I had left the value in for confirmation :(

Reply:
To call it only once, is it better to do it in plugin/luasnip.vim?

Reply:
Aaah, alright :D

Reply:
Mhm, I would require users to just add that line before loading.
I don't really like it being there if it isn't necessary.

Reply:
Ah, Just let the user write the whole thing.

Reply:
`s/write/copy-paste/` :P

Reply:
But yes, that was my idea